### PR TITLE
fixes for later Nim version

### PR DESCRIPTION
**Pull request by ba0f3**

### CODE REVIEW

File: src/keystone.nim [hunk 1]
```diff
@@ -92,7 +92,7 @@ type
   ##  Resolver callback to provide value for a missing symbol in @symbol.
   ##  To handle a symbol, the resolver must put value of the symbol in @value,
   ##  then returns True.
-  SymbolResolver* = proc (symbol: cstring; value: ptr csize): bool {.cdecl.}
+  SymbolResolver* = proc (symbol: cstring; value: ptr csize_t): bool {.cdecl.}
```
Comment (71) [inline src/keystone.nim:95]: I wasn't aware of the difference between `csize` and `csize_t`. Thanks for the fix.

File: src/keystone.nim [hunk 2]
```diff
@@ -72,7 +72,7 @@ type
     MISSINGFEATURE,
     MNEMONICFAIL
 
-  KeystoneError* = object of Exception
+  KeystoneError* = object of CatchableError
```
Comment (71) [inline src/keystone.nim:75]: I wasn't aware of the difference between `Exception` and `CatchableError`. Thanks for the fix.

File: src/keystone.nim [hunk 3]
```diff
@@ -3,10 +3,10 @@ import strutils
 when defined(windows):
   const libname = "keystone.dll"
 else:
-  const libname = "keystone"
-
+  const libname = "libkeystone.so"
```
Comment (71) [inline src/keystone.nim:6]: I'm gonna trust you on this.

File: src/keystone.nim [hunk 4]
```diff
@@ -142,19 +142,20 @@ template sparc64*(m: type Mode): Mode = (1 shl 3).Mode
 ## SparcV9 mode.
 template v9*(m: type Mode): Mode = (1 shl 4).Mode
 
-proc `or`*(a, b: Mode): Mode {.inline.} =
-  (a.uint or b.uint).Mode
+proc `or`*(a, b: Mode): uint {.inline.} =
```
Comment (71) [inline src/keystone.nim]: Why use `uint` (or `Mode`) here and in the rest of the document?

**Review by 71 — COMMENTED** [review]
Thanks for the PR! I'm mostly wondering about the change regarding `Mode`. Also, can the minimum Nim version be lowered?

File: src/keystone.nim (hunk 1 above)
Comment (ba0f3) [reply src/keystone.nim:95]: its just changed recently, and `csize` is deprecated

File: src/keystone.nim (hunk 2 above)
Comment (ba0f3) [reply src/keystone.nim:75]: `Exception` is deprecated in favor of `CatchableError` or `Defect` too

File: src/keystone.nim (hunk 4 above)
Comment (ba0f3) [reply src/keystone.nim]: `or`  operator between Modes causes out of range error, but I dont know why, just a little dirty hack here to make it works

**Comment by ba0f3** [comment]
actually it works with Nim version 1.2.8 too

File: src/keystone.nim (hunk 4 above)
Comment (71) [reply src/keystone.nim]: Then in that case I think `Mode` should be a `distinct uint`, and its values `littleEndian`, `bigEndian`, and `b{16,32,64}` should be defined like `template xxx*(m: type Mode): Mode = (1 shl xxx).Mode`

Comment (ba0f3) [reply src/keystone.nim]: yeah, that's a good idea. Im going to make the changes now

File: src/keystone.nim [hunk 5]
```diff
@@ -102,15 +102,18 @@ type
...
-type Mode* {.pure.} = enum
-  littleEndian = 0 ##  Little-endian mode (default).
-
-  b16 = 1 shl 1    ##  16-bit mode.
-  b32 = 1 shl 2    ##  32-bit mode.
-  b64 = 1 shl 3    ##  64-bit mode.
-
-  bigEndian = 1 shl 30  ##  Big-endian mode.
-
+type Mode* {.pure.} = distinct uint
```
Comment (71) [inline src/keystone.nim]: The type no longer needs `{.pure.}`.

**Comment by 71** [comment]
Thanks!

**Pull request merged**